docs/DEPLOYMENT.md

```diff
@@ -0,0 +1,410 @@
+## Setting up the ports webapp
+
+For testing and development purposes, the ports webapp can be deployed and populated locally. The main Django app is contained
+inside the `app` directory in the root of the project.
+
+###### Project structure
+```
+|-- app     (main django app)
+|-- config
+|-- docs
+|-- .gitignore
+|-- .gitmodules
+|-- .travis.yml
+|-- Dockerfile
+|-- env.sample
+|-- LICENSE
+|-- README.md
+```
+
+As mentioned, the `app` directory is a standalone django app. However, the app has been Dockerised and all the pain of
+setting up a local environment can be avoided when working with Docker. But in case you do not want to work with Docker,
+we will also walk through the standard process of setting up the app.
+
+## Getting the dependencies ready
+
+The external dependencies of the app are: PostgreSQL, Solr and Memcached. The search page(/search/) is completely derived from
+Solr. Memcached is optional, and can be avoided if you don't want to cache the results of stats and trac tickets.
+
+If you are new to these technologies, you may use the instructions provided below, otherwise you are free to install and setup these
+dependencies using your preferred mechanism.
+
+### PostgreSQL
+
+Download and install `Postgres.app` from [here](https://postgresapp.com/downloads.html).
+
+After this, connect as root:
+```bash
+su - postgres
+```
+
+Create your user(replace username and password with your preferred credentials):
+```bash
+CREATE USER username PASSWORD 'passowrd';
+```
+
+Create a new database;
+
+```bash
+CREATE DATABASE webapp;
+```
+
+Grant permissions to your user on this database.
+
+```bash 
+GRANT ALL PRIVILEGES ON DATABASE webapp TO username;
```

Comment:
This is ok for playing around, but it's not suitable for the final deployment. I'm not sure whether we need to be picky on this.